test/checkChallenge.test.js

```diff
@@ -0,0 +1,1627 @@
+      
+      assert.deepEqual(result, ({ reject: true,   msg: 'This bot only accepts games from 13k players or stronger ranking.' }));
+
+    });
+
+    it('accept user ranking edge min', () => {
+
+      const notification = base_challenge({ ranked: false, user: { ranking: 17 } }); // "13k"
+
+      config.minrank = 17;
+      config.maxrank = 32;
+      
+      const result = conn.checkChallengeMandatory(notification);
+      
+      assert.deepEqual(result, ({ reject: false }));
+
+    });
+
+    it('accept user ranking between min and max', () => {
+
+      const notification = base_challenge({ ranked: false, user: { ranking: 25 } }); // "5k"
+
+      config.minrank = 17;
+      config.maxrank = 32;
+      
+      const result = conn.checkChallengeMandatory(notification);
+      
+      assert.deepEqual(result, ({ reject: false }));
+
+    });
+
+    it('accept user ranking edge max', () => {
+      const notification = base_challenge({ ranked: false, user: { ranking: 32 } }); // "3d"
+
+      config.minrank = 17;
+      config.maxrank = 32;
+      
+      const result = conn.checkChallengeMandatory(notification);
+      
+      assert.deepEqual(result, ({ reject: false }));
+
+    });
+
+    it('reject user ranking too high', () => {
+
+      const notification = base_challenge({ ranked: false, user: { ranking: 35 } }); // "6d"
+
+      config.minrank = 17;
+      config.maxrank = 32;
+      
+      const result = conn.checkChallengeMandatory(notification);
+      
+      assert.deepEqual(result, ({ reject: true,   msg: 'This bot only accepts games from 3d players or weaker ranking.' }));
+
+    });
+
+    it('reject user ranking too high (9d+)', () => {
+
+      const notification = base_challenge({ ranked: false, user: { ranking: 41 } }); // "12d"
+
+      config.minrank = 17;
+      config.maxrank = 32;
+      
+      const result = conn.checkChallengeMandatory(notification);
+      
+      assert.deepEqual(result, ({ reject: true,   msg: 'This bot only accepts games from 3d players or weaker ranking.' }));
+
+    });
+
+    it('reject user ranking too high (pro)', () => {
+
+      const notification = base_challenge({ ranked: false, user: { ranking: 37 } }); // "1p" (1p" = "8d")
+
+      config.minrank = 17;
+      config.maxrank = 32;
+      
+      const result = conn.checkChallengeMandatory(notification);
+      
+      assert.deepEqual(result, ({ reject: true,   msg: 'This bot only accepts games from 3d players or weaker ranking.' }));
+   });
+   
+  });
+
+  describe('Min Max Handicap', () => {
+
+    it('reject handicap too low (automatic handicap is -1)', () => {
+
+      const notification = base_challenge({ ranked: false, handicap: -1 });
+
+      config.minhandicap = 0;
+      config.maxhandicap = 2;
+      
+      const result = conn.checkChallengeMinMax(notification);
+      
+      assert.deepEqual(result, ({ reject: true,   msg: 'Minimum number of handicap stones is 0, please increase the number of handicap stones.' }));
```

Comment:
yes, i assumed it would be clear that the user needs to change handicap stones but actually it's a total mess, user may just think there is a bug with the handicap not being recognized as "automatic" and keep trying again

it's really similar to the no non-square boardsizes case though , so we may try to find a common path for both maybe

Reply:
actually for this a way simpler and consistent approach may be to export in config if minhandicap > -1 exports.noautohandicap = true;
same as what was done in #304